src/index.ts

```diff
@@ -76,4 +77,70 @@ app.post("/logout", async(c) => {
   return new Response();
 });
 
+app.post("/attendance", async(c) => {
+  /*ネットワークの確認を行う*/
+  const allowPattern = new RegExp(c.env.ALLOW_IP_REGEX);
+  if ( !allowPattern.test(c.req.header("cf-connectiong-ip") as string) ) {
+    return c.text("Forbidden", 403);
+  }
+
+  /*学生の認証を行う*/
+  //sessionがcookieあるか確認
+  const session = c.get("session");
+  if ( !session ){
+    return c.text("Unauthorized", 401);
+  }
+  //session情報がsessionテーブルにある確認
+  const existResult = await c.env.DB
+  .prepare("SELECT * FROM session WHERE id = ?")
+  .bind(session['id'])
+  .first();
+  const isExist = existResult == null;
+  if ( isExist ){
+    return c.text("Unauthorized", 401);
+  }
+  //roleが学生か確認
+  const isStudent = await c.env.DB
+  .prepare("SELECT * FROM account WHERE role = STUDENT AND id = ?")
+  .bind(session['account_id'])
+  .first();
+  if ( isStudent ){
+    return c.text("Unauthorized", 401);
+  }
```

Comment:
```suggestion
  const accountEntry = await c.env.DB
  .prepare("SELECT * FROM account WHERE role = 'STUDENT' AND id = ?")
  .bind(session.account_id)
  .first();
  const isStudent = accountEntry !== null;
  if (!isStudent) {
    return c.text("Unauthorized", 401);
  }
```
変数の意味が逆になっており, falsy も誤検知しかねないので `null` の取り扱いとして不適切です